app/src/main/java/com/example/myapplication/FirstFragment.java

```diff
@@ -97,6 +95,7 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
             @Override
             public void afterTextChanged(Editable s) {
                 MainActivity.teamNumber = binding.input.getText() + "";
+                UserModel.getMatchData().setTeamNumber(Integer.parseInt(MainActivity.teamNumber));
```

Comment:
This may throw an exception if the value entered is not a valid number - we should validate (or try...catching) it.

Reply:
Nevermind, the widget is taking care of it